base/socket.jl

```diff
@@ -148,6 +148,11 @@ end
 
 # Parsing
 
+const ipv4_leading_zero_error = """
+Leading zeros in IPv4 addresses are disallowed due to ambiguity.
+If octal or hexadecimal, convert to decimal, otherwise remove leading zero.
```

Comment:
IMO this reads kind of weird, since the first sentence is a complete sentence and this one isn't. Perhaps the second could be rephrased as
> If the address is in octal or hexadecimal, convert it to decimal, otherwise remove the leading zero.

?